### DISCUSSION TITLE
How to verify consecutive lead times

### DISCUSSION THREAD

**Discussion by rvalenzuelar** [dtcenter/METplus#1032] [Configuration] [answered]
Greetings,

I am looking to perform a GridStat analysis for consecutive lead times of precipitation forecast. 

For example, I have the following forecast files (6-h precip accumulations):

gfs.0p25.2021012618.f102.grib2
gfs.0p25.2021012700.f096.grib2
gfs.0p25.2021012706.f090.grib2
...
gfs.0p25.2021013006.f018.grib2
gfs.0p25.2021013012.f012.grib2
gfs.0p25.2021013018.f006.grib2

that are valid at 2021-01-31T0000

I processed the rain gauges observations with point2grid, so I have a precip_obs_2021013100.nc file.

What would be the most efficient workflow in this case? Running GridStat for each forecast/obs file pair? Should I use pcp_combine to combine the forecast files and then run GridStat? What would be the correct config that would allow to run this?

Best,

Raúl

**Comment by j-opatz** [comment, marked as answer]
Hi Raúl, and thank you for your question.

For your particular use case, you would need to run GridStat over each matched pair. PCPCombine is used in precipitation use cases if a user wants to aggregate the forecast precipitation time to the same accumulation time as the observation data set; for example, if you had a 24 hour accumulation observation and 4, 6-hour accumulation forecast files for that same period of time. Because each of your forecast files is valid for the same time, you do not want to combine those forecasted amounts. 

However, there is an automated method to accomplish this goal. Rather than running GridStat directly dozens of times, I strongly suggest looking into the METplus wrappers. There is a Python wrapped version of each of the MET tools which allows users like yourself to specify how many times to run a tool by valid and initialization times. Take a look [here,](https://metplus.readthedocs.io/en/develop/index.html) which is a link to the METplus User's Guide. In particular, I'll share this simple grid_stat [use case ](https://metplus.readthedocs.io/en/develop/generated/met_tool_wrapper/GridStat/GridStat.html#sphx-glr-generated-met-tool-wrapper-gridstat-gridstat-py)as a starting point. Your configuration file will look a little different since your forecast and observation files are in netCDF (the use case has grib format forecast files), but you can see how the FCST_GRID_STAT_INPUT_TEMPLATE variable is used to loop over multiple files with varying lead times.

**Reply by rvalenzuelar** [reply]
Thanks @j-opatz! I will give it a try and report back.

**Reply by rvalenzuelar** [reply]
Hi @j-opatz, after following and reading the METplus tutorial, and the links you shared, there is one thing that is not working. 

In my experiment I have a set of lead times all forecasting to the same valid time (2021-01-31T0000). Currently, I have my time configuration as follows:

LOOP_BY = INIT
INIT_TIME_FMT = %Y%m%d%H
INIT_BEG = 2021012618
INIT_END = 2021012700
INIT_INCREMENT = 6H
LEAD_SEQ = 102H, 96H (there are more, but I am testing with these two)

The problem is that METplus is searching the following init times:
2021012618.f102 (OK)
2021012618.f096 (doesn't exist)--> error
2021012700.f102 (doesn't exist)-->error
2021012700.f096 (OK)

instead of looping:
2021012618.f102
2021012700.f096

What should be the correct time configuration?

**Reply by rvalenzuelar** [reply]
Just found the correct way. I will write it here to keep track of the workflow.

Since I have only 1 valid time the configuration should be:

LOOP_BY = VALID
VALID_TIME_FMT = %Y%m%d%H
VALID_BEG = 2021013100
VALID_END = 2021013106
VALID_INCREMENT = 12H
LEAD_SEQ = 102H, 96H

According to the documentation ([here](https://metplus.readthedocs.io/en/latest/Users_Guide/systemconfiguration.html?highlight=skipping#skipping-times)) with this configuration the VALID_INCREMENT value will make the running process to exceed the VALID_END time, so the verification will be made only for the VALID_BEG time. In addition, providing all the corresponding lead times in LEAD_SEQ will make the all the runs with correct init times.
